convert_hf_to_gguf.py

```diff
@@ -186,6 +186,7 @@ def map_tensor_name(self, name: str, try_suffixes: Sequence[str] = (".weight", "
         return new_name
 
     def set_gguf_parameters(self):
+        self.gguf_writer.add_type(gguf.GGUFType.MODEL)
```

Comment:
Some models completely override `set_gguf_parameters`, so the type won't be set for them.

Reply:
I moved this line to `Model.__init__` --> https://github.com/ggerganov/llama.cpp/pull/8332/commits/b1c40695029f4317a8c1598d1523631b60755b44

Reply:
`Model.__init__` is also used by `convert_lora_to_gguf.py`, so I think it might be broken by this due to duplicate key for the model type.